src/auth.py

```diff
@@ -34,6 +37,23 @@ def wrapper(*args, **kwargs):
 	return wrapper
 
 
+def require_admin_status(func):
+	"""
+	A route decorator that give Forbidden error if the user is not an admin of this
+	course.
+	:param func: a route function that takes NetID and course id as the first two positional
+		arguments.
+	"""
+	@wraps(func)
+	def wrapper(*args, **kwargs):
+		netid = kwargs[UID_KEY]
+		cid = kwargs[CID_KEY]
+		if not verify_staff(netid, cid) or not verify_admin(netid, cid):
```

Comment:
Is it possible to not be staff, but be admin?

Reply:
In our current system, no. Every logic I could see assumes that admin is a subset of staff. This doesn't really make sense to me. Because we never use the role when a person is an admin but not a staff.

Reply:
Admin is a subset of staff.

Reply:
I think what @PradyumnaShome is implying is: if it is not possible for a person to be an admin and not a staff, then just doing:
```
if not verify_admin(netid, cid):
```
should be sufficient.

Reply:
As discussed offline, currently the admins and staff are separate sets in the DB and they can theoretically diverge. So, for now, we need to check both.